rest/stats_context.go

```diff
@@ -0,0 +1,54 @@
+package rest
+
+import (
+	"time"
+	"os"
+	"github.com/elastic/gosigar"
+)
+
+// Group the stats related context that is associated w/ a ServerContext into a struct
+type statsContext struct {
+	statsLoggingTicker *time.Ticker
+	cpuStatsSnapshot   *cpuStatsSnapshot
+}
+
+// A snapshot of the cpu stats that are used for stats calculation
+type cpuStatsSnapshot struct {
+
+	// The cumulative CPU time that's been used in various categories, in units of jiffies (clock ticks).
+	// This spans all processes on the system.
+	totalTimeJiffies uint64
+
+	// CPU time spent in user code, measured in clock ticks.  Only applies to this process.
+	procUserTimeJiffies uint64
+
+	// CPU time spent in kernel code, measured in clock ticks.  Only applies to this process.
+	procSystemTimeJiffies uint64
+}
+
```

Comment:
As stated above, I think it would be a lot cleaner if we encapsulated the (relatively complex) cpu snapshot comparison logic inside statsContext.

Reply:
Fixed